src/history/history.go

```diff
@@ -94,18 +94,21 @@ func StoreHealth(hc *HealthCheck) {
 	go writeHealth(hc)
 }
 
-// GetState returns the current state of the service
-func GetState(repoURL string) State {
+// GetState returns the current state of the service and a tag identifying it
+func GetState(repoURL string) (State, string) {
 	mux.Lock()
 	defer mux.Unlock()
 	if _, ok := history[repoURL]; !ok {
 		history[repoURL] = NewService()
 	}
-	return *history[repoURL].Current
+	tag := makeTag(repoURL, *history[repoURL].Current)
+	return *history[repoURL].Current, tag
 }
 
 // SetState sets the current state of service
-func SetState(repoURL string, cur State) error {
+// Compares the current state of repoURL to provided tag, and if it matches,
+// sets it to cur
+func SetState(repoURL string, tag string, cur State) error {
```

Comment:
I think we should change the variable `cur` to `reqState`. `cur` confues with the current state